alcs-api/src/application-amendment/application-amendment.service.ts

```diff
@@ -0,0 +1,185 @@
+      region: true,
+      localGovernment: true,
+      decisionMeetings: true,
+    },
+    card: {
+      board: true,
+      type: true,
+      status: true,
+      assignee: true,
+    },
+  };
+
+  getByBoardCode(boardCode: string) {
+    return this.getBy({ card: { board: { code: boardCode } } });
+  }
+
+  getByApplication(applicationFileNumber: string) {
+    return this.getBy({ application: { fileNumber: applicationFileNumber } });
+  }
+
+  getBy(findOptions: FindOptionsWhere<ApplicationAmendment>) {
+    return this.amendmentRepository.find({
+      where: findOptions,
+      relations: this.DEFAULT_RELATIONS,
+    });
+  }
+
+  mapToDtos(
+    amendments: ApplicationAmendment[],
+  ): Promise<ApplicationAmendmentDto[]> {
+    return this.mapper.mapArrayAsync(
+      amendments,
+      ApplicationAmendment,
+      ApplicationAmendmentDto,
+    );
+  }
+
+  async create(createDto: ApplicationAmendmentCreateDto, board: Board) {
+    const amendment = new ApplicationAmendment({
+      submittedDate: new Date(createDto.submittedDate),
+      isTimeExtension: createDto.isTimeExtension,
+    });
+
+    amendment.card = await this.cardService.create('AMEND', board, false);
+    amendment.application = await this.getOrCreateApplication(createDto);
+
+    const savedAmendment = await this.amendmentRepository.save(amendment);
+    return this.getByUuid(savedAmendment.uuid);
+  }
+
+  private async getOrCreateApplication(
+    createDto: ApplicationAmendmentCreateDto,
+  ) {
+    const existingApplication = await this.applicationService.get(
+      createDto.applicationFileNumber,
+    );
+
+    if (existingApplication) {
+      return existingApplication;
+    } else {
+      return await this.applicationService.create(
+        {
+          fileNumber: createDto.applicationFileNumber,
+          typeCode: createDto.applicationTypeCode,
+          regionCode: createDto.regionCode,
+          localGovernmentUuid: createDto.localGovernmentUuid,
+          applicant: createDto.applicant,
+        },
+        false,
+      );
+    }
+  }
+
+  async update(uuid: string, updateDto: ApplicationAmendmentUpdateDto) {
+    const existingAmendment = await this.getByUuidOrFail(uuid);
+
+    existingAmendment.reviewDate = formatIncomingDate(updateDto.reviewDate);
+    existingAmendment.submittedDate = formatIncomingDate(
+      updateDto.submittedDate,
+    );
+    existingAmendment.isReviewApproved = updateDto.isReviewApproved;
+    existingAmendment.isTimeExtension = updateDto.isTimeExtension;
+
+    const amendment = await this.amendmentRepository.save(existingAmendment);
+
+    return this.getByUuid(amendment.uuid);
+  }
+
+  async delete(uuid: string) {
+    const amendment = await this.getByUuidOrFail(uuid);
+    return this.amendmentRepository.softRemove([amendment]);
+  }
+
+  private async getByUuidOrFail(uuid: string) {
+    const amendment = await this.amendmentRepository.findOneByOrFail({
+      uuid,
+    });
+
+    if (!amendment) {
```

Comment:
I think this will not execute. The line above should throw an exception if there is no amendment

Reply:
Ah good point, changed to findOneBy and throw this exception so we can control the exception type and error.